sshd-core/src/main/java/org/apache/sshd/client/auth/pubkey/UserAuthPublicKey.java

```diff
@@ -321,10 +331,19 @@ protected boolean processAuthDataRequest(ClientSession session, String service,
                                                                              + ByteArrayBuffer.DEFAULT_SIZE + Long.SIZE);
         buffer.putString(username);
         buffer.putString(service);
-        buffer.putString(name);
+        Integer hostBoundPubKeyVersion = session.getAttribute(DefaultClientKexExtensionHandler.HOSTBOUND_AUTHENTICATION);
+        boolean doHostBoundAuth = hostBoundPubKeyVersion != null && hostBoundPubKeyVersion.intValue() == 0;
+        if (doHostBoundAuth) {
+            buffer.putString(HostBoundPubkeyAuthentication.AUTH_NAME);
+        } else {
+            buffer.putString(name);
+        }
         buffer.putBoolean(true);
         buffer.putString(algo);
         buffer.putPublicKey(pubKey);
+        if (doHostBoundAuth) {
```

Comment:
I think the log message should include the correctly used algorithm name if *doHostBoundAuth* is *true*

Reply:
Done.